### PR TITLE
DEPR: deprecated nonkeyword arguments in to_parquet

### PR DESCRIPTION
**Pull request by rsm-23**
- [x] xref #54229 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v2.2.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by rsm-23** [comment]
@mroeschke can you please help me understand what's causing this error?
>AssertionError: Warning not set with correct stacklevel. File where warning is raised: /home/runner/micromamba/envs/test/lib/python3.9/site-packages/pyarrow/pandas_compat.py != /home/runner/work/pandas/pandas/pandas/tests/io/test_parquet.py. Warning message: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.

**Comment by mroeschke** [comment]
You can set `check_stacklevel=False` for your test. Seems like a 3rd party warning is messing up the stacklevel checking

**Comment by rsm-23** [comment]
@mroeschke still getting some error like this. Can you please help on this?

> pandas/tests/io/test_parquet.py::test_parquet_pos_args_deprecation - ImportError: Unable to find a usable engine; tried using: 'pyarrow', 'fastparquet'.
A suitable version of pyarrow or fastparquet is required for parquet support.

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -359,6 +359,18 @@ def test_cross_engine_fp_pa(df_cross_compat, pa, fp):
...
+def test_parquet_pos_args_deprecation():
+    # GH-54229
+    df = pd.DataFrame({"a": [1, 2, 3]})
+    msg = (
+        r"Starting with pandas version 3.0 all arguments of to_parquet except for the "
+        r"argument 'path' will be keyword-only."
+    )
+    with tm.assert_produces_warning(FutureWarning, match=msg, check_stacklevel=False):
+        buffer = BytesIO()
+        df.to_parquet(buffer, "auto")
```
Comment (mroeschke) [inline pandas/tests/io/test_parquet.py:371]:
```suggestion
def test_parquet_pos_args_deprecation(engine):
    # GH-54229
    df = pd.DataFrame({"a": [1, 2, 3]})
    msg = (
        r"Starting with pandas version 3.0 all arguments of to_parquet except for the "
        r"argument 'path' will be keyword-only."
    )
    with tm.assert_produces_warning(FutureWarning, match=msg, check_stacklevel=False):
        buffer = BytesIO()
        df.to_parquet(buffer, engine)
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rsm-23

**Pull request merged**